src/buildercore/bootstrap.py

```diff
@@ -486,7 +486,15 @@ def _update_ec2_node():
         # write out environment config (/etc/cfn-info.json) so Salt can read CFN outputs
         write_environment_info(stackname, overwrite=True)
 
+        # bit of a hack, but project config merging doesn't apply to top-level values
+        # in this case we look for an alternate salt version under a project's 'ec2' section
         salt_version = context['project']['salt']
+        alt_salt_version = context['ec2'].get('salt')
+
+        # and we only use it if we're going masterless
```

Comment:
why the restriction to masterless? Could eventually be relaxed?

Reply:
it would require the master server to be running the same version of Salt. I don't recall any problems with mismatches in Salt versions between client and server but we've always done it with master first.

it could definitely be relaxed if we wanted to test that aspect as well